### ISSUE TITLE
miner.coinbase interprets Address as Number

### ISSUE THREAD

**Issue by robmcl4** [ConsenSys-archive/ganache#2364]
Environment:

```
docker container: Ubuntu 20.04
node v17.4.0
yarn run v1.22.17
ganache v7.0.2 (@ganache/cli: 0.1.3, @ganache/core: 0.1.3)
```

What I ran:

```
yarn ganache --fork.url ws://127.0.0.1:8546 --miner.coinbase 0xA0a0a0A0A0A0a0a0A0A0a0A0a0A0a0A0A0A0a0a0
```

What I expected: ganache starts with address `0xA0a0a0A0A0A0a0a0A0A0a0A0a0A0a0A0A0A0a0a0` as the coinbase address.

What actually happened (output, exits with code 1)

```
yarn run v1.22.17
warning package.json: No license field
$ /opt/project/node_modules/.bin/ganache --fork.url ws://127.0.0.1:8546 --miner.coinbase 0xA0a0a0A0A0A0a0a0A0A0a0A0a0A0a0A0A0A0a0a0
ganache v7.0.2 (@ganache/cli: 0.1.3, @ganache/core: 0.1.3)
invalid coinbase address index: 9.170206351880175e+47
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Note that 0xA0a0a0A0A0A0a0a0A0A0a0A0a0A0a0A0A0A0a0a0 = 9.17...e+47. The command does work if I place a whitespace character at the end of the coinbase, to force interpretation as a string. It does _not_ work if the whitespace character is prepended to the address.

```
root@ca6b1eff69bb:/opt/project# yarn ganache --fork.url ws://127.0.0.1:8546 --miner.coinbase '0xA0a0a0A0A0A0a0a0A0A0a0A0a0A0a0A0A0A0a0a0 '
yarn run v1.22.17
warning package.json: No license field
$ /opt/project/node_modules/.bin/ganache --fork.url ws://127.0.0.1:8546 --miner.coinbase '0xA0a0a0A0A0A0a0a0A0A0a0A0a0A0a0A0A0A0a0a0 '
ganache v7.0.2 (@ganache/cli: 0.1.3, @ganache/core: 0.1.3)
Starting RPC server

Available Accounts
==================
(0) 0x7a3175F719a229e92C24d1cC8046e43a48f78a89 (1000 ETH)
(1) 0x89569dcA032AB8cAA5833f4610b90DA305938138 (1000 ETH)
(2) 0x7767A2effFDd8Fd8b733FE1A9369B2656DccF3cA (1000 ETH)
(3) 0x889AA163852AFF8a213a4d21E7E831BA841f4271 (1000 ETH)
(4) 0xd7769BDc0d281E700a2bf855D834814F49ed4a61 (1000 ETH)
(5) 0x896A1A1F02438BA225e7cB25fA16e3c3559D7e9B (1000 ETH)
(6) 0x0d8f7E938537f775C2306477D9D40CB6F9A203f2 (1000 ETH)
(7) 0xb211c1920B256701Ba0F61C8E9593DCD27135b41 (1000 ETH)
(8) 0x6dD238034B40d57076E2Db3CD108554480cE1cAc (1000 ETH)
(9) 0x973C4C9aC10C15D874dEbD552084A10feDDb0E15 (1000 ETH)

Private Keys
==================
(0) 0x8365d5447191bb8b59d7c3c036947ca976633548f341f9951fe6cc083c7c0bcb
(1) 0xa2cb894958b02f379bbeff2f6d85a64576cc4ad5d4822aab36df3352a8cdef4e
(2) 0xc74dae1690d37dc8ddb369a2671f6860bdbb2f19aa22283ad841ea503324b8cd
(3) 0x398eae1140d63795c5bcc20a0b6a7e770154cded5d5bfe01f80d78e9c8a5bffa
(4) 0x778f9e15716daf40d459c7ff6f168b1e51ade13f451d12138b938b709039df4f
(5) 0xa371b504fb6b954cae2b523cbc1b4b66f51912ae35d24cadc2fe4616d57f97eb
(6) 0xbaf2d908c1775a0029d4b76c35d916b1e6ddfb4efc0a1f09e972d598092bfd05
(7) 0xef4d2f54c6ee6569ea1674bea15fe1067cdf082170881b5ec569415f8c6da1ee
(8) 0x7a46816227d3a4d841030f897574247316a60715a9d2e42171f025465ab69b18
(9) 0x4912cc09087435a7ecd5d415d945bf10ef548aad1bc3a974333c3e15fd6e9516

HD Wallet
==================
Mnemonic:      donkey rookie roof cream guilt inmate invite safe spatial song bring cannon
Base HD Path:  m/44'/60'/0'/0/{account_index}

Default Gas Price
==================
2000000000

BlockGas Limit
==================
30000000

Call Gas Limit
==================
50000000

Forked Chain
==================
Location:        ws://127.0.0.1:8546/
Block:           14202293
Network ID:      1
Time:            Mon Feb 14 2022 05:18:21 GMT+0000 (Coordinated Universal Time)

Chain Id
==================
1337

RPC Listening on 127.0.0.1:8545
```

**Comment by gnidan** [comment]
Thanks for bringing this up! Sorry about this!

Adding this to our backlog to resolve.

**Comment by robmcl4** [comment]
Closing as this was fixed in [v7.0.3](https://github.com/trufflesuite/ganache/releases/tag/v7.0.3)
